### DISCUSSION TITLE
ScriptReference/Rigidbody.SweepTestAll

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#402] [Notes]
# ScriptReference/Rigidbody.SweepTestAll

https://docs.unity3d.com/ScriptReference/Rigidbody.SweepTestAll.html

**Comment by docscomments** [comment]
**Comment by `@uvivagabond`:**  

The collision matrix influences which colliders will be taken into account. The layer assigned to rigidbody determines which colliders are detected with. If colliders in certain layers do not interact, they will not be detected by this method either.

**Comment by ggevrin** [comment]
Does not detect colliders already in contact (even if they are in the sweep direction)
